sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ThinClientStoreModel.java

```diff
@@ -50,6 +51,15 @@ public ThinClientStoreModel(
             globalEndpointManager,
             httpClient,
             ApiType.SQL);
+
+        String userAgent = userAgentContainer != null
+            ? userAgentContainer.getUserAgent()
+            : UserAgentContainer.BASE_USER_AGENT_STRING;
+
+        final ImmutableMap.Builder<String, String> builder = ImmutableMap.builderWithExpectedSize(1);
+        this.defaultHeaders = builder
+            .put(HttpConstants.HttpHeaders.USER_AGENT, userAgent)
+            .build();
```

Comment:
Given that it appears this is an immutable map with size 1, why not use `Collections.singletonMap` which optimizes as effectively a key-value pair.